src/v/raft/consensus.cc

```diff
@@ -127,6 +127,7 @@ ss::future<> consensus::stop() {
     _vote_timeout.cancel();
```

Comment:
can you please add some high-level functional description of this change to the commit message? it sounds like the result of this change is that the recovery stm for relaxed consistency situations is never stopped, but that isn't clearly stated as an important result of this change, if it happens to be true.

Reply:
is an alternative design to using a condition variable to signal disk appends to the recovery stm to instead identify the point where, you write:

_This may lead to the situation where we
would introduce unnecessary latency before delivering data to the
follower as we would have to wait for next follower heartbeat response
before starting next recovery process._

and then re-launch the recovery stm at this point instead of waiting for the heartbeat? i'm assuming that the cost of actually starting the stm is negligible. otherwise we have put a condition variable broadcast into the hot path for all users (but it is also extremely cheap to broadcast to a cv with no waiters).

Reply:
Proposed alternative sounds reasonable but it would require few lookups in the follower state map to prevent recoveries to be started twice, also since recovery doesn't work under the mutex it may lead to race condition where we would have two STM's running.

Reply:
Do you mean a general PR comment or commit message of this single commit ? in this commit message it is stated that recovery STM doesn't stop:
```
r/recovery_stm: do not stop recovery when follower is up to date

Waiting for next disk append instead of stopping recovery process when
data are replicated with relaxed consistency.
```